pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/typedefinition/JavaTypeDefinition.java

```diff
@@ -18,19 +18,20 @@
 
     private final TypeDefinitionType definitionType;
 
-    public enum TypeDefinitionType {
-        EXACT, UPPER_BOUND, UPPER_WILDCARD, LOWER_WILDCARD
-    }
-
     protected JavaTypeDefinition(TypeDefinitionType definitionType) {
         this.definitionType = definitionType;
     }
 
-    public static JavaTypeDefinition forClass(TypeDefinitionType type, List<JavaTypeDefinition> intersectionTypes) {
+    public static JavaTypeDefinition forClass(TypeDefinitionType type, Class<?> clazz,
```

Comment:
who is using this new factory method?

Reply:
It's for convenience, should we minimize the number of factory methods?

Reply:
not really worried about it, just couldn't find any calls to it, so I wondered if it was needed at all / it should really be public (ie: if only called from child classes, it could be package-private)

Reply:
Hmm, it is used in the ClassTypeResolver and the ClassTypeResolverTest classes.

Reply:
awesome, must have missed it.